src/app/modules/registration/components/thumbnails/thumbnails.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnChanges, Input, ChangeDetectionStrategy, OnInit } from '@angular/core';
+import { map, distinctUntilChanged, Observable, ReplaySubject, combineLatest } from 'rxjs';
+import {
+  AttachmentUploadEditModel,
+  AttachmentUploadEditModelWithBlob,
+  ExistingOrNewAttachment
+} from 'src/app/modules/common-registration/registration.models';
+import { NewAttachmentService } from 'src/app/modules/common-registration/registration.services';
+import { RemoteOrLocalAttachmentEditModel } from 'src/app/core/services/draft/draft-model';
+import { attachmentsComparator } from 'src/app/core/helpers/attachment-comparator';
+
+/**
+ * Show thumbnails of all images for given registration.
+ */
+@Component({
+  selector: 'app-thumbnails',
+  templateUrl: './thumbnails.component.html',
+  styleUrls: ['./thumbnails.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ThumbnailsComponent implements OnChanges, OnInit {
+  @Input() attachments: ExistingOrNewAttachment[]; //attachments for given draft registration
+  @Input() draftUuid: string;
+
+  private attachmentsSubject = new ReplaySubject<ExistingOrNewAttachment[]>(1);
```

Comment:
Er det nødvendig å bruke en replaysubject her? replaysubject brukes som regel når man må legge til rette for at noe starter med å lytte "for seint" osv, men siden denne er privat og bare lyttes på av ting inne i komponentet holder det kanskje med en vanlig subject?

Reply:
Godt poeng. Takk for påminnelsen!

Reply:
Hvis jeg bytter til vanlig Subject, vises ikke bilder hvis jeg ikke har fylt ut noe i skjemaet (altså valgt et ikon).
Så jeg beholder ReplySubject av denne grunn. Sikkert andre måter å løse dette på, men tenker at det er godt nok.